soc/adi/max32/Kconfig

```diff
@@ -0,0 +1,29 @@
+# Analog Devices MAX32xxx MCU family
+
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_MAX32
+	select ARM
+	select CPU_HAS_ARM_MPU
+	select CPU_HAS_FPU
+	select CPU_CORTEX_M_HAS_SYSTICK
+	select CLOCK_CONTROL
+	select BUILD_OUTPUT_HEX
+
+config SOC_MAX32690
+	select CPU_CORTEX_M4
+
+if SOC_FAMILY_MAX32
+
+config MAX32_ON_ENTER_CPU_IDLE_HOOK
+	bool "CPU idle hook enable"
+	default y
```

Comment:
last question: why is this default y? are you preventing sleep by default? maybe default y if DEBUG or something similar would make more sense

Reply:
Thanks for suggestion,
This parameters requires to access device incase of it goes to sleep mode immediately after power up. So that it requires both for debugging and FW update.
We are working on our OpenOCD fork, trying to handle this case. We are trying to add feature that allow to access MCU even it goes to sleep mode. It almost done. After we release new version of ADI openocd distro we will revisit MAX32_ON_ENTER_CPU_IDLE_HOOK flag. 
So that for new keeping it as it is would be better for MAX32 MCUs.